### PR TITLE
Fix #4972 Shape center and size not reflective of actual

### LINKED ISSUE #4972: 20221130 Center and sizes for STL objects not reflected accurately in Radia
Object center is dependent on how how the STL is exported, sometimes isn't aligned with the shape. Can likely be fixed by using trimesh.center_mass (or .centroid) to determine appropriate center.

Object size is currently just default values, can likely use trimesh bounding box to determine actual size.

### PR DESCRIPTION
**Pull request by Benbenbenin0**
Merged #4971 into this PR along with #4972

Shapes are now built at center provided by dragging in the design tab, also now displays shapes using the same STL file.

Shape size is now updated to the actual after building from the imported file, however the design tab still leaves default values in the "size [mm]" field that do nothing when changed.

@mkeilman think we need a new issue for that? Not sure whether to let the user scale the STL's size in the designer, remove the field for STL's or do something else.

### CODE REVIEW

**Comment by mkeilman** [comment]
> Shape size is now updated to the actual after building from the imported file, however the design tab still leaves default values in the "size [mm]" field that do nothing when changed.
> 
> @mkeilman think we need a new issue for that? Not sure whether to let the user scale the STL's size in the designer, remove the field for STL's or do something else.

We should still display the size but disable the input fields. See https://github.com/radiasoft/sirepo/blob/dcb249c0625782ee3cf8ed4877244a6b437f1b27/sirepo/package_data/static/js/radia.js#L3841
It's a one-line fix if you want to add it to this PR, but a new issue is fine too

File: sirepo/template/radia_util.py [hunk 1]
```diff
@@ -304,6 +304,7 @@ def build_stl(**kwargs):
     g_id = radia.ObjPolyhdr(
         d.vertices, (numpy.array(d.faces) + 1).tolist(), d.magnetization
     )
+    g_id = radia.TrfOrnt(g_id, radia.TrfTrsl([d.center[0], d.center[1], d.center[2]]))
```
Comment (mkeilman) [inline sirepo/template/radia_util.py:307]: This doesn't quite work on its own. We want to use the center defined by the user, irrespective of the points in the stl file. I think the easiest way is to subtract off the stl's center first:
```python
o.stlCentroid = mesh.centroid.tolist()
...
g_id = radia_util.build_stl(
...
    center=[x - o.stlCentroid[i] for i, x in enumerate(o.center)],
)
```

**Review by mkeilman — CHANGES_REQUESTED** [review]
See comment on centering

File: sirepo/template/radia_util.py (hunk 1 above)
Comment (Benbenbenin0) [reply sirepo/template/radia_util.py:307]: Ah, I was testing radia.TrfOrnt() without realizing I was using (0,0,0) as my initial center, misunderstood the function.

File: sirepo/package_data/static/js/radia.js [hunk 2]
```diff
@@ -3870,9 +3870,16 @@ SIREPO.viewLogic('geomObjectView', function(appState, panelState, radiaService,
         panelState.showField('geomObject', 'materialFile', o.material === 'custom');
 
         panelState.enableField('geomObject', 'size', true);
+
+        if (o.type == 'stl') {
+            panelState.enableField('geomObject', 'size', false);
+            //TODO(BG): Only disables 'size' field, need to build shape to get sizes to update values (likely will need to send request since python)
```
Comment (mkeilman) [inline sirepo/package_data/static/js/radia.js:3876]: I'll make a separate issue for this

**Review by mkeilman — COMMENTED** [review]
One minor comment

**Comment by mkeilman** [comment]
ping @Benbenbenin0 - a couple of changes still in queue here. If you need to focus on ldap that's ok - I'll fix this up; lmk one way or the other

**Review by mkeilman — APPROVED** [review]

**Comment by Benbenbenin0** [comment]
@mkeilman Thanks for the fixes and merge

**Pull request merged**